opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/exponential/aggregation.py

```diff
@@ -0,0 +1,555 @@
+
+    def _swap(self, other: "_ExponentialBucketHistogramAggregation") -> None:
+
+        for attribute in [
+            "_positive",
+            "_negative",
+            "_sum",
+            "_count",
+            "_zero_count",
+            "_min",
+            "_max",
+            "_mapping",
+        ]:
+            temp = getattr(self, attribute)
+            setattr(self, attribute, getattr(other, attribute))
+            setattr(other, attribute, temp)
+
+    def _copy_into(
+        self, other: "_ExponentialBucketHistogramAggregation"
+    ) -> None:
+        other._clear()
+
+        for attribute in [
+            "_positive",
+            "_negative",
+            "_sum",
+            "_count",
+            "_zero_count",
+            "_min",
+            "_max",
+            "_mapping",
+        ]:
+            setattr(other, attribute, getattr(self, attribute))
+
+    def _update_by_incr(self, number: Union[int, float], incr: int) -> None:
+
+        value = float(number)
+
+        if self._count == 0:
+            self._min = number
+            self._max = number
+
+        else:
+            if number < self._min:
+                self._min = number
+            if number > self._max:
+                self._max = number
+
+        self._count += incr
+
+        if value == 0:
+            self._zero_count += incr
+            return
+
+        self._sum += number * incr
+
+        if value > 0:
+            buckets = self._positive
+        else:
+            value = -value
+            buckets = self._negative
+
+        self._update(buckets, value, incr)
+
+    def _downscale(self, change: int) -> None:
+        """
+        Subtracts change from the current mapping scale
+        """
+
+        if change == 0:
+            return
+
+        if change < 0:
+            raise Exception(f"Impossible change of scale: {change}")
+
+        new_scale = self._mapping.scale - change
+
+        self._positive.downscale(change)
+        self._negative.downscale(change)
+
+        if new_scale <= 0:
+            mapping = ExponentMapping(new_scale)
+        else:
+            mapping = LogarithmExponentialHistogramMapping(new_scale)
+
+        self._mapping = mapping
+
+    # pylint: disable=no-self-use
+    def _change_scale(self, high: int, low: int, size: int) -> int:
+        """
+        Calculates how much downscaling is needed by shifting the high and low
+        values until they are separated by no more than size.
+        """
+
+        change = 0
+
+        # FIXME this slightly different from the Go implementation. It should
+        # be functionally equal but avoids an infinite loop in certain
+        # circumstances (high == 0, low == -1, size == 1).
+        while high - low > size:
+            high = high >> 1
+            low = low >> 1
+
+            change += 1
+
+        if high - low == size:
+            high = high >> 1
+            low = low >> 1
+
+            change += 1
+
+        return change
```

Comment:
```suggestion
        while high - low >= size:
            high = high >> 1
            low = low >> 1

            change += 1
        return change
```
The minimum max-size should be 2; does that help?  See the original: https://github.com/lightstep/otel-launcher-go/blob/main/lightstep/sdk/metric/aggregator/histogram/structure/exponential.go#L349